### ISSUE TITLE
defaultScalariformSettings option do not turn off automatic code formatting in Compile task

### ISSUE THREAD

**Issue by kristerr** [sbt/sbt-scalariform#48]
According to doc:

"If you don't want sbt to automatically format your source files when the tasks compile or test:compile, just add defaultScalariformSettings instead of scalariformSettings to your build definition....
.sbt build example:
import com.typesafe.sbt.SbtScalariform
SbtScalariform.defaultScalariformSettings
"""

It does not work.

**Comment by ryan-williams** [comment]
I saw this as well, seems like you need to explicitly disable the plugin as well! The following works for me:

```scala
import scalariform.formatter.preferences._
import com.typesafe.sbt.SbtScalariform
import com.typesafe.sbt.SbtScalariform.ScalariformKeys

disablePlugins(SbtScalariform)

SbtScalariform.defaultScalariformSettings

ScalariformKeys.preferences := ScalariformKeys.preferences.value
                               .setPreference(AlignSingleLineCaseStatements, true)
                               .setPreference(DoubleIndentClassDeclaration, true)
                               .setPreference(DanglingCloseParenthesis, Preserve)
```

**Comment by klaeufer** [comment]
I have included the plugin in `~/.sbt/0.13/plugins/build.sbt` and these settings in `~/.sbt/0.13/global.sbt`. 

None of these attempts to disable automatic formatting works for me. The only way I can get it to work is if I put `disablePlugins(SbtScalariform)` in my project.

Looks I'll have to remove the plugin from the global configuration and add it to each project without putting it in version control.

**Comment by godenji** [comment]
See updated [readme](https://github.com/sbt/sbt-scalariform/blob/master/README.md). 

tl;dr; use `formatOnDemandSettings` in place of `defaultScalariformSettings` to enable opt-in formatting.
